server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java

```diff
@@ -2165,12 +2174,43 @@ public void testStats() throws Exception {
         // total
         assertStats(afterForthRequestStats.totalStats(), 4, 0, 0);
         // pipeline
-        assertPipelineStats(afterForthRequestStats.pipelineStats(), "_id1", 3, 0, 0);
-        assertPipelineStats(afterForthRequestStats.pipelineStats(), "_id2", 1, 0, 0);
+        assertPipelineStats(afterForthRequestStats.pipelineStats(), "_id1", 3, 0, 0, startSize1, endSize1);
+        assertPipelineStats(afterForthRequestStats.pipelineStats(), "_id2", 1, 0, 0, startSize2, endSize2);
```

Comment:
This comment made me aware that there might be a bug around here because I'm not sure these values are actually supposed to be equal, I think it just turns out that way because the mock processors aren't modifying the data so there's no increase in bytes.

But I dug into this a little bit more and realized I misremembered this part of the test as this test https://github.com/elastic/elasticsearch/blob/12b31e79caa67fd717ffa49166eff54e179acca9/server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java#L181 when we brought up testing a failure
And I think parts of this test don't behave the way I thought it did (in the 3rd test the byte stats values for `_id1` hasn't increased, and in the 4th with the failure `startSize1` is equal to `endSize1`), so I'm going to dig into this a bit more

Reply:
Figured out what was wrong - when the stats of a pipeline gets updated via another pipeline's stats it wasn't updating the `IngestPipelineMetric` stats, so I fixed that and changed some of the tests as necessary

Reply:
Yeah that was part of what led me to investigating that section, I didn't realize at first that `endSize1` was increasing there